extensions/amp-animation/0.1/web-animations.js

```diff
@@ -1179,15 +1193,41 @@ class CssContextImpl {
 
   /** @override */
   getViewportSize() {
+    this.resolveViewportRect_();
     if (!this.viewportSize_) {
+      const {width, height} = this.viewportRect_;
       this.viewportSize_ = {
-        width: this.win_./*OK*/ innerWidth,
-        height: this.win_./*OK*/ innerHeight,
+        width: width * this.scaleFactorX_,
+        height: height * this.scaleFactorY_,
       };
     }
     return this.viewportSize_;
   }
 
+  /** @private */
+  resolveViewportRect_() {
+    if (this.viewportRect_) {
+      return;
+    }
+    if (!this.scope_) {
+      this.viewportRect_ = layoutRectLtwh(
+        0,
+        0,
+        this.win_./*OK*/ innerWidth,
+        this.win_./*OK*/ innerHeight
+      );
+      return;
+    }
+    this.viewportRect_ = this.scope_./*OK*/ getBoundingClientRect();
+
```

Comment:
Let's do `const {offsetWidth, offsetHeight} = this.scope_` - this would be easier to see all measures in the same place.